cartoframes/data/dataframe_dataset.py

```diff
@@ -0,0 +1,231 @@
+import pandas as pd
+from warnings import warn
+from tqdm import tqdm
+
+from carto.exceptions import CartoException, CartoRateLimitException
+
+from .dataset_base import DatasetBase
+from ..columns import Column, normalize_names
+from .utils import decode_geometry, compute_geodataframe, detect_encoding_type, map_geom_type
+
+# avoid _lock issue: https://github.com/tqdm/tqdm/issues/457
+tqdm(disable=True, total=0)  # initialise internal lock
+
+
+class DataFrameDataset(DatasetBase):
+    def __init__(self, data, context=None, schema=None):
+        super(DataFrameDataset, self).__init__()
+
+        self._df = data
+
+    @property
+    def dataframe(self):
+        """Dataset DataFrame"""
+        return self._df
+
+    def get_geodataframe(self):
+        """Converts DataFrame into GeoDataFrame if possible"""
+        gdf = compute_geodataframe(self)
+        if not gdf.empty:
+            self._df = gdf
+
+        return self._df
+
+    @property
+    def dataset_info(self):
+        raise CartoException('We can not extract Dataset info from a query. Use `Dataset.from_table()` method '
```

Comment:
This should be written from the `dataframe` perspective instead of `query`. I'm also curious if this can be inherited instead of being defined here again?

Reply:
Copy paste error :) 

We could check it in the Dataset class (the same happens with several methods) ensuring the proper strategy class, but we would break the strategy pattern. In the other side, we will need to add methods "without functionality" like this one.

Reply:
After writting the previous one, I realised it is possible to ask about `dataset_info` in a `DataFrameDataset` if the dataset is already uploaded. Working on it

Reply:
Thank you @andy-esch good catch!